examples/renameimage.py

```diff
@@ -38,44 +97,100 @@
+
+    if new_image:
+        master_image = new_image
+    elif old_image:
+        master_image = old_image
+        old_image.name = sys.argv[2]
+        glancesync.update_metadata_image(master, old_image)
+        print 'Renamed image in master region'
+    else:
+        msg = 'Neither an image with the old name nor a one with the new name'\
+            ' exists in the master region'
+        logging.error(msg)
+        sys.exit(-1)
+
     if len(sys.argv) > 3:
         regions = sys.argv[3:]
     else:
-        regions = glancesync.get_regions(False)
+        regions = glancesync.get_regions()
 
     for region in regions:
         print region + ' ',
         try:
             images = glancesync.get_images_region(region)
-            # only rename if the target name does not exist and the
-            # source name is unique.
+
+            # only rename if the target name does not exist, the image
+            # has the right checksum and there is only one image to
+            # rename.
             image_to_rename = None
             destination_name_exists = False
+            ignore_region = False
+            image_already_renamed = None
             for image in images:
-                if image.name == sys.argv[1]:
+                if image.name == sys.argv[1] and\
+                        image.checksum == master_image.checksum:
                     if image_to_rename is not None:
                         print 'Not renamed.'
                         msg = 'Name {0} is not unique in region {1}'
                         logging.error(msg.format(image.name, region))
-
+                        ignore_region = True
                         break
                     else:
                         image_to_rename = image
                 if image.name == sys.argv[2]:
-                    # No error yet. Perhaps this image has been already renamed
-                    destination_name_exists = True
+                    image_already_renamed = image
 
-            if image_to_rename is not None:
-                if destination_name_exists:
+            if ignore_region:
+                continue
+
+            if image_to_rename:
+                if image_already_renamed:
                     print 'Not renamed.'
                     msg = 'Destination name {0} already exists in region {1}'
                     logging.error(msg.format(sys.argv[2], region))
                 else:
-                    image_to_rename.name = sys.argv[2]
-                    glancesync.update_metadata_image(region, image_to_rename)
+                    update_image(image_to_rename, master_image)
                     print 'Renamed'
             else:
-                print 'Not found.'
+                if image_already_renamed:
+                    if metadata_outdated(image_already_renamed, master_image):
+                        update_image(image_already_renamed, master_image)
+                        print 'medatadata updated in already renamed image'
```

Comment:
This sentence will produce a technical debt, please avoid usage of print statement and change to:

print('medatadata updated in already renamed image')